core/tests/webdriverio_utils/waitFor.js

```diff
@@ -0,0 +1,228 @@
+ */
+var presenceOf = async function(element, errorMessage) {
+  await browser.waitUntil(
+    await until.presenceOf(element),
+    {
+      timeout: DEFAULT_WAIT_TIME_MSECS,
+      timeoutMsg: errorMessage
+    });
+};
+
+/**
+ * @param {Object} element - Element expected to be present in the DOM and has
+ *                           height and width that is greater than 0.
+ * @param {string} errorMessage - Error message when element is invisible.
+ */
+var visibilityOf = async function(element, errorMessage) {
+  await browser.waitUntil(
+    await until.visibilityOf(element),
+    {
+      timeout: DEFAULT_WAIT_TIME_MSECS,
+      timeoutMsg: errorMessage
+    });
+};
+
+/**
+ * @param {Object} element - Element who attribute we are waiting to
+ *                           for
+ * @param {Object} attribute - Name of attribute
+ * @param {Object} value - Value we are waiting attribute to have
+ * @param {Object} errorMessage - Error message in case wait times out
+ */
+var elementAttributeToBe = async function(
+    element, attribute, value, errorMessage
+) {
+  await browser.waitUntil(async function() {
+    return await element.getAttribute(attribute) === value;
+  },
+  {
+    timeout: DEFAULT_WAIT_TIME_MSECS,
+    timeoutMsg: errorMessage
+  });
+};
+
+var visibilityOfInfoToast = async function(errorMessage) {
+  var toastInfoElement = $('.toast-info');
+  await visibilityOf(toastInfoElement, errorMessage);
+};
+
+var invisibilityOfInfoToast = async function(errorMessage) {
+  var toastInfoElement = $('.toast-info');
+  await invisibilityOf(toastInfoElement, errorMessage);
+};
+
+var invisibilityOfLoadingMessage = async function(errorMessage) {
+  var loadingMessage = $('.e2e-test-loading-message');
+  await invisibilityOf(loadingMessage, errorMessage);
+};
+
+var visibilityOfSuccessToast = async function(errorMessage) {
+  var toastSuccessElement = $('.toast-success');
+  await visibilityOf(toastSuccessElement, errorMessage);
+};
+
+var fadeInToComplete = async function(element, errorMessage) {
+  await visibilityOf(element, 'Editor taking too long to appear');
+  await browser.waitUntil(async function() {
+    return (await element.getCSSProperty('opacity')).value === 1;
+  },
+  {
+    timeout: DEFAULT_WAIT_TIME_MSECS,
+    timeoutMsg: errorMessage
+  });
+};
+
+var modalPopupToAppear = async function() {
+  await visibilityOf(
+    $('.modal-body'), 'Modal taking too long to appear.');
+};
+
+/**
+ * Check if a file has been downloaded
+ */
+var fileToBeDownloaded = async function(filename) {
+  var name = Constants.DOWNLOAD_PATH + '/' + filename;
+  await browser.waitUntil(function() {
+    return fs.existsSync(name);
+  },
+  {
+    timeout: DEFAULT_WAIT_TIME_MSECS,
+    timeoutMsg: 'File was not downloaded!'
+  });
+};
+
+
+var clientSideRedirection = async function(
+    action, check, waitForCallerSpecifiedConditions) {
+  // Action triggering redirection.
```

Comment:
We do not have any substitute available for waitForAngularEnabled in webdriverio. This function if set to false, Protractor will not wait for Angular $http and $timeout tasks to complete before interacting with the browser. So to make webdriverio code work I increased the wait time to 30000 ms in order to give time to $http and $timeout task to complete on redirection.

Reply:
Where have you updated "wait time to 30000 ms"?

Reply:
PTAL at line no 205 of waitFor.js file

Reply:
Based on our discussion in google chats there will be no difference in the waiting time. It will remain the same i.e 10000ms. 
Webdriver doesn't wait for angular to be present on the page as compared to protractor. Hence, we don't need a substitute for waitforAngularEnabled().